sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -229,33 +229,59 @@ case class RLike(left: Expression, right: Expression) extends StringRegexExpress
 
 
 /**
- * Splits str around pat (pattern is a regular expression).
+ * Splits str around pattern (pattern is a regular expression).
```

Comment:
pattern? regex? we should use a consisntent word.

Reply:
going to switch to `regex`, makes more sense given that with the use of `pattern` we always have to define it as a regex